protos/hardware/hardware.proto

```diff
@@ -65,63 +71,162 @@ service HardwareService {
   };
 }
 
+/*
+ * PushRequest is the body for the Push method. It contains information about
+ * an hardware.
+ */
 message PushRequest {
+  /*
+   * hardware describes how the hardware you want to register to Tinkerbell looks like.
+   * Hostname, mac address, DHCP, network interfaces, metadata and so on.
+   */
   Hardware data = 1;
 }
 
+/*
+ * Empty represents an empty response
+ */
 message Empty {
 }
 
+/*
+ * GetRequest is the request used by the Get various get methods exposed by the
+ * hardware serice.
+ */
 message GetRequest {
   string mac = 1;
   string ip = 2;
   string id = 3;
 }
 
+/*
+ * Hardware describes a specific device managed by Tinkerbell
+ */
 message Hardware {
+  /*
+   * DHCP represents
+   */
   message DHCP {
     message IP {
       string address = 1;
       string netmask = 2;
       string gateway = 3;
       int64 family = 4;
     }
+    /*
+     * The mac address for your hardware.  It is important to get it
+     * right because it is used from Tinkerbell to dispatch workflows to the
+     * right destination.
+     */
     string mac = 1;
-    reserved 2; // obsolete ip
+    reserved 2;
```

Comment:
can we keep the `// obsolete` tombstones just so we have a clear record of what was deprecated / reason why the number is reserved?